File: 1547-destination-city/destination-city.js
/**
 * @param {string[][]} paths
 * @return {string}
 */
var destCity = function(paths) {
    let set = new Set();
    paths.forEach((path) => {
        set.add(path[0]);
        set.add(path[1]);
    });
    paths.forEach((path) => {
        set.delete(path[0]);
    });
    let ans = '';
    set.forEach((s) => ans = s);
    return ans;
};

File: 1547-destination-city/destination-city.java
class Solution {
    public String destCity(List<List<String>> paths) {
        var set = new HashSet<String>();
        for(var path : paths) {
            set.add(path.get(0));
            set.add(path.get(1));
        }
        for(var path : paths) {
            set.remove(path.get(0));
        }
        return set.iterator().next();
    }
}